lucille-core/src/main/java/com/kmwllc/lucille/connector/storageclient/LocalStorageClient.java

```diff
@@ -74,19 +82,10 @@ public void shutdown() throws IOException {
 
   @Override
   public void traverse(Publisher publisher) throws Exception {
-    try (Stream<Path> paths = Files.walk(startingDirectoryPath).sorted()) {
-      paths.filter(this::isValidPath)
-          .forEachOrdered(path -> {
-            Path fullPath = path.toAbsolutePath().normalize();
-            String fullPathStr = fullPath.toString();
-            String fileExtension = FilenameUtils.getExtension(fullPathStr);
-            tryProcessAndPublishFile(publisher, fullPathStr, fileExtension, new FileReference(fullPath));
-          });
-    } catch (InvalidPathException e) {
-      throw new ConnectorException("Path string provided cannot be converted to a Path.", e);
-    } catch (SecurityException | IOException e) {
-      throw new ConnectorException("Error while traversing file system.", e);
-    }
+    
+    // LocalStorage class implements the FileVisitor so we can just walk it.
+    Files.walkFileTree(startingDirectoryPath, this);
```

Comment:
the interface here needs to pass a FileVisitor, so when a file is visited, it's getting the LocalStorageClient which implements the visitFile method.  The result is that LocalStorageClient needs a handle to the publisher.  Otherwise we would need to make the publisher implement FileVisitor.

Reply:
1) The change here removes `paths.filter(this::isValidPath)`. That test is how we support includes and excludes. If that is removed without being replaced elsewhere, tests should fail, otherwise we have a gap in test coverage.

2) We could have a wrapper around Publisher that implements FileVisitor. I'm not convinced we need to store the Publisher or modify Publisher itself.